src/parser.ts

```diff
@@ -355,10 +355,20 @@ const EXPR_BUILDER: MarkVisitor<NodeTypes.ExprNode> = {
       return result
     }
 
-    const args: NodeTypes.ExprNode[] = []
+    let args: NodeTypes.ExprNode[] = []
+
     while (p.getMark().name !== 'func_args_end') {
-      args.push(p.process(EXPR_BUILDER))
+      if (argumentShouldBeSelector(namespace, name, args)) {
+        // Since the diff/delta functions aren't validated yet we only want to validate the selector
+        // being used. We expect the null valued arg to throw an error at evaluation time.
+        p.process(SELECTOR_BUILDER)
+        args.push(NodeTypes.SelectorNode)
```

Comment:
Doesn't this have to be like this (since `SelectorNode` is an interface)?

```suggestion
        args.push({type: 'Selector'})
```